baybe/searchspace/discrete.py

```diff
@@ -255,16 +255,24 @@ def from_product(
         # Set defaults
         constraints = constraints or []
 
-        # Create a dataframe representing the experimental search space
-        exp_rep = parameter_cartesian_prod_to_df(parameters)
+        # Use Polars for Cartesian product of parameters and apply constraint
```

Comment:
Not sure if we've decided to add it in this PR already, but the optionality of polars is missing